packages/amplify-util-mock/src/utils/lambda/hydrate-env-vars.ts

```diff
@@ -0,0 +1,59 @@
+export type Resource = {
+  resourceName: string;
+  category: string;
+  output: Record<string, string>;
+};
+export type Resources = Resource[];
+export type ApiDetails = {
+  apiName: string;
+  apiKey: string;
+  apiId: string;
+  url: string;
+  dataSources: {
+    name: string;
+    Arn: string;
+    type: string;
+  }[];
+};
+export function hydrateAllEnvVars(
+  resources: Resources,
+  sourceEnvVars: Record<string, string>,
+  apiDetails?: ApiDetails,
+): Record<string, string> {
+  const outputMap: Record<string, object> = resources.reduce((acc: Record<string, object>, r) => {
+    const category = r.category;
+    const resourceName = r.resourceName.toLowerCase();
+    const outputs = Object.entries(r.output || {}).reduce((sum, [name, value]) => {
```

Comment:
Nit: these nested reduce functions are a bit confusing to reason through. Maybe consider putting them into helper functions?